meteor/sofie-server/data-access/repositories/cache/cached-rundown-repository.ts

```diff
@@ -0,0 +1,48 @@
+import { RundownRepository } from '../interfaces/rundown-repository'
+import { Rundown } from '../../../model/entities/rundown'
+import { MisconfigurationException } from '../../../model/exceptions/misconfiguration-exception'
+import { Identifier } from '../../../model/interfaces/identifier'
+
+export class CachedRundownRepository implements RundownRepository {
+	private static instance: RundownRepository
+
+	public static getInstance(rundownRepository?: RundownRepository): RundownRepository {
+		if (!this.instance) {
+			if (!rundownRepository) {
+				throw new MisconfigurationException(
+					`No RundownRepository provided. Unable to create instance of ${CachedRundownRepository.name}`
+				)
+			}
```

Comment:
The singletons are still needed. But since we control who calls `getInstance` through the facade the parameter is no longer optional and the `getInstance` method has been simplified.